src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -294,6 +294,21 @@ private void PrepareProjectExecutables()
             if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(DebugSessionPortVar)))
             {
                 exeSpec.ExecutionType = ExecutionType.IDE;
+
+                string? launchProfileName = project.SelectLaunchProfileName();
+                if (!string.IsNullOrEmpty(launchProfileName))
+                {
+                    var launchProfile = project.GetEffectiveLaunchProfile();
+                    if (launchProfile is not null && !string.IsNullOrWhiteSpace(launchProfile.CommandLineArgs))
+                    {
+                        var cmdArgs = launchProfile.CommandLineArgs.Split((string?)null, StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+                        if (cmdArgs is not null && cmdArgs.Length > 0)
+                        {
+                            exeSpec.Args = [];
+                            exeSpec.Args.AddRange(cmdArgs);
+                        }
+                    }
```

Comment:
This logic is duplicated, we should do it once and reuse it if it's already been done. (resolving the effective launch profile)

Reply:
I guess we also need to make sure that this doesn't break anything? It's not clear why this resolution is only in the else block.

@karolz-ms @mitchdenny

Reply:
I guess this is a branch so we can't share the resolution but we can put it in a method